### ISSUE TITLE
CU80 RGB MATRIX - Brightness Limits

### ISSUE THREAD

**Issue by maz0r** [qmk/qmk_firmware#12165]
Sorry, I missed this PR and saw it only after it was merged, and now I see some potential problems with it, which you may want to fix with a separate PR.

The brightness limiting that was done with `#define RGBLIGHT_LIMIT_VAL 50` is not handled by the RGB Matrix code — the proper option for RGB Matrix would be `RGB_MATRIX_MAXIMUM_BRIGHTNESS`. This may be important to avoid power issues.

(Note that `rgb_matrix_set_color()` bypasses the limit set by `RGB_MATRIX_MAXIMUM_BRIGHTNESS`, but that would probably be OK for the usage here, because only 2 indicator LEDs would be at the full brightness.)

LED flags are not 100% correct — 1 (`LED_FLAG_MODIFIER`) should be set for LEDs under the modifier keys like Ctrl, Shift, Alt (some effects use that flag to color the modifier keys differently from the normal ones). Setting this flag for Caps Lock and Scroll Lock looks wrong, and technically it is not even needed, because `rgb_matrix_indicators_kb()` will override the color of those LEDs irrespective of the flags used for them (except that you don't override their colors when the corresponding indicators are off, so in that case the current RGB Matrix effect would show there).

.....

BTW, speaking of keymaps, having `RGB_SPI` and `RGB_SPD` keycodes somewhere would be useful for RGB Matrix (in a pinch you can get by with just `RGB_SPI`, because holding Shift will reverse the direction of parameter change for RGB keycodes).

_Originally posted by @sigprof in https://github.com/qmk/qmk_firmware/issues/12019#issuecomment-792740732_

**Comment by maz0r** [comment]
Adding as issue, as I'm going to fix this tonight after work :)

**Comment by maz0r** [comment]
This was also included, but I believe these to be more enchantments than issues at this time:


....

(Also I wonder whether the `rgb_matrix_indicators_kb()` API is really good — the issue is that there is no way to override that function from the keymap, so if someone wants those LEDs to display the RGB Matrix effect instead of working as indicators, there would be no clean way to disable the indicator code apart from patching the keyboard-level code. But that looks more like a higher-level problem which is not really specific to this particular keyboard. You could move the indicator code to the keymap level, but that would leave the configurator users without those indicators, and you might need to duplicate that code in the `via` keymap too. Another way to hack around this problem is a custom `#ifdef` around `rgb_matrix_indicators_kb()`, but it does not look like a proper general solution.)

In addition, I think that a better way to add the RGB Matrix support would be by modifying the existing `capsunlocked/cu80/v2_iso` and `capsunlocked/cu80/v2_ansi` keyboards by adding the RGB Matrix code there inside `#ifdef RGB_MATRIX_ENABLE`, and then adding `capsunlocked/cu80/v2_iso/rgb` and `capsunlocked/cu80/v2_ansi/rgb` models with just some changed `rules.mk` options and no real code:
```makefile
RGBLIGHT_ENABLE = no
RGB_MATRIX_ENABLE = yes
```
Doing it this way would avoid the code duplication (even the keymaps could be reused). But this might be too late to fix now, and other maintainers may have different opinions.

(One issue here is that VIA would require different `PRODUCT_ID` values for RGB Lighting and RGB Matrix variants of the firmware, and I'm not sure whether having both values inside `#ifdef RGB_MATRIX_ENABLE` in the same `config.h` would not cause problems with some scripts.)

....

_Originally posted by @sigprof in https://github.com/qmk/qmk_firmware/issues/12019#issuecomment-792740732_
